src/sync/initial.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * @module sync
+ */
+
+import assert from "assert";
+import PeerInfo from "peer-info";
+
+import {IBeaconDb} from "../db";
+import {IBeaconChain} from "../chain";
+import {SyncRpc} from "./rpc";
+import {INetwork} from "../network";
+import {ReputationStore} from "./reputation";
+import {ILogger} from "../logger";
+import {SLOTS_PER_EPOCH} from "../constants";
+import {BeaconBlockHeadersResponse, BeaconBlockBodiesResponse, BeaconBlock} from "../types";
+
+export class InitialSync {
+  private db: IBeaconDb;
+  private chain: IBeaconChain;
+  private rpc: SyncRpc;
+  private network: INetwork;
+  private reps: ReputationStore;
+  private logger: ILogger;
+  public constructor(opts, {db, chain, rpc, network, reps, logger}) {
+    this.db = db;
+    this.chain = chain;
+    this.rpc = rpc;
+    this.network = network;
+    this.reps = reps;
+    this.logger = logger;
+  }
+  public async syncToPeers(): Promise<void> {
+    // Sort peers from best to worst
+    const peers = this.network.getPeers().sort((peerA, peerB) => {
```

Comment:
It seems we would sort because often, how about we expose method in network called getSortedPeers to network can implement sort depending on implementation

Reply:
I think this is the only place we're currently sorting peers. There are several metrics to sort peers on, (best slot, best reputation, etc), so I'm hesitant to put this on the network interface.